src/types/error.rs

```diff
@@ -77,7 +77,9 @@ pub(crate) fn convert_panics<T, F: UnwindSafe + FnOnce() -> NeonResult<T>>(f: F)
             };
             let (data, len) = Utf8::from(&msg[..]).truncate().lower();
             unsafe {
-                neon_runtime::error::throw_error_from_utf8(data, len);
+                #[cfg(feature = "napi-runtime")]
+                neon_runtime::error::clear_exception(env.to_raw());
```

Comment:
The existing API defines panics as taking priority over exceptions. However, in `N-API` you may only throw an exception when not in a throwing state.